src/mindlessgen/molecules/miscellaneous.py

```diff
@@ -81,6 +85,29 @@ def set_random_charge(ati: np.ndarray, verbosity: int = 1) -> tuple[int, int]:
         return charge, uhf
 
 
+def calculate_protons(natoms: np.ndarray) -> int:
+    """
+    Calculate the number of protons in a molecule from the atom list.
+    """
+    protons = 0
+    for i, atom_count in enumerate(natoms):
+        protons += atom_count * (i + 1)
+    return protons
+
+
+def calculate_ligand_protons(natoms: np.ndarray, nel: int) -> int:
+    """
+    Calculate the number of ligand protons in a molecule if lanthanides or actinides are within the molecule.
+    """
+    f_protons = sum(
+        atom * (i - 2)
+        for i, atom in enumerate(natoms)
```

Comment:
Shouldn't this be `for ati, occurrence` or similar? And why do we have this `i - 2`?

Reply:
Following the same logic as in `set random charge` this (i-2) comes from the fact that the Ln( or Ac) is charged 3+ (f_protons - 3) and to correct the 0 based indexing we need to add  + 1.

Reply:
I add a comment to clarify this.

Reply:
I although change the name to ligand_electrons

Reply:
Okay, but what about the variable names `i` and `atom`?

Reply:
Ah yes i change them to your suggestion